pkg/controllerutils/controllerutil.go

```diff
@@ -169,3 +169,11 @@ func FindRbacCondition(conditions []managedv1alpha1.Condition, conditionType man
 	}
 	return nil
 }
+
+// check if namespace exist and NamespacePhase is non terminating
+func ValidateNamespace(namespace *corev1.Namespace) bool {
+	if namespace.Name != "" && string(namespace.Status.Phase) != "Terminating" {
```

Comment:
could be better represented by 
```suggestion
	if namespace.Name != "" && namespace.Status.Phase != corev1.NamespaceTerminating {
```